src/vm/instruction.ts

```diff
@@ -248,101 +254,120 @@ export function decodeInstruction(
     UnsignedInteger.uint16And(flags, pcUpdateMask),
     pcUpdateOff
   );
-  var pcUpdate: PcUpdate;
+  let pcUpdate: PcUpdate;
   switch (pcUpdateNum) {
     case 0:
+      // pc = pc + instruction size
       pcUpdate = PcUpdate.PcUpdateRegular;
       break;
     case 1:
+      // pc = res
       pcUpdate = PcUpdate.PcUpdateJump;
       break;
     case 2:
+      // pc = pc + res
       pcUpdate = PcUpdate.PcUpdateJumpRel;
       break;
     case 4:
+      // if dst != 0 then pc = pc + instruction_size else pc + op1
       pcUpdate = PcUpdate.PcUpdateJnz;
       break;
     default:
-      return new Err(InvalidOp1Src);
+      return new Err(InvalidPcUpdate);
   }
 
   const resLogicNum = UnsignedInteger.uint16Shr(
     UnsignedInteger.uint16And(flags, resLogicMask),
     resLogicOff
   );
-  var resLogic: ResLogic;
+  let resLogic: ResLogic;
   switch (resLogicNum) {
     case 0:
+      // jnz with res_logic == 0 and pc_update == 3 then unconstrained
```

Comment:
could you add a bit more information?
I don't understand the use of the word "with" here,
what does unconstrained mean?

what does res = op1 mean? maybe if we use a simple sentence it'll helper the reader

Reply:
the issue is that we are only decoding here, not executing yet, but for example `res = op1` I think means `result = operand 1`. Can't be sure, this comes from [cairo paper](https://eprint.iacr.org/2021/1063.pdf#page=33). `with` means if you have `jnz` and res_logic == 0 and pc_update == 3, then the resLogic is unconstrained, meaning you don't care about its value.